### ISSUE TITLE
[Android] Style with alignText property "right" error

### ISSUE THREAD

**Issue by mrKlar** [facebook/react-native#2702]
When using a style property of textAlign: "right", I get the following error on Android.  It is working in iOS.

<img width="762" alt="screen shot 2015-09-14 at 11 08 23 pm" src="https://cloud.githubusercontent.com/assets/1790018/9866828/8545d98e-5b35-11e5-9f73-915e79ce66d2.png">

**Comment by ide** [comment]
cc @mkonicek I can repro this with center alignment too.

**Comment by kmagiera** [comment]
This is for TextInput right? According to the docs `textAlign` is android only http://facebook.github.io/react-native/docs/textinput.html#content (there is a typo in platform annotation)
Allowed values are: 'start', 'center', 'end'
So the error you're getting is not a good description of the actual problem but you essentially are providing invalid value for that property (flow should be able to catch this earlier as prop type is annotated)

**Comment by ide** [comment]
@kmagiera I encountered this error in a `Text` component where `textAlign` works on iOS. The problem is that when you write `<Text style={{textAlign: 'center'}}>`, "center" is not converted to an enum before sending it to Java.

**Comment by kmagiera** [comment]
there is discrepancy in the way we handle this for Text and InputText. For Text we just use strings on android so there is no need for the value to be converted. Redbox here is cause by native code trying to read number while string is provided. Are you also getting a redbox because of that with Text component, if so can you send a stacktrace?

**Comment by ide** [comment]
@kmagiera oh you're right. this was a TextInput component (`textAlign: center` does work on iOS though).

**Comment by sowdri** [comment]
Is there any workaround for this?

**Comment by kmagiera** [comment]
Just use allowed values: 'center', 'start' or 'end'
see http://facebook.github.io/react-native/docs/textinput.html#content

**Comment by mrKlar** [comment]
<img width="762" alt="screen shot 2015-10-04 at 3 06 09 pm" src="https://cloud.githubusercontent.com/assets/1790018/10269554/7872308a-6aa9-11e5-80a1-1720820fa712.png">

Hi still get an error, here are my dependencies

 "dependencies": {
    "react-native": "^0.11.4",
    "react-native-custom-actsheet": "0.0.4",
    "react-native-popover": "^0.2.0",
    "react-native-responsive-image": "^1.0.0",
    "numeral": "^1.5.3",
    "react-native-i18n-complete":"^0.1.2"
  }

**Comment by mrKlar** [comment]
I got it to work... BUT, my error is that I was trying to use the property in the CSS.  It does not work there, you must use it as an attribute of TextInput directly.  Home it is going to work in CSS someday.

**Comment by theoriginalgri** [comment]
Interestingly this also affects iOS somehow.

My code is:
```
  render() {
    var textAlign = Platform.OS == 'ios' ? 'right' : 'end';
    return (
        <TextInput textAlign={textAlign} ref='input' />
    );
  }
```

This works well (but feels wrong) on both platforms but iOS presents a warning:
```
2015-12-09 12:52:27.062 [warn][tid:com.facebook.React.JavaScript] 'Warning: Failed propType: Invalid prop `textAlign` of value `right` supplied to `TextInput`, expected one of ["start","center","end"]. Check the render method of `TestComponent`.'
```

If the value is changed to 'end' for iOS, too, RN 0.16.0 throws an exception:
```
Invalid NSTextAlignment 'end'. should be one of: (auto, center, justify, left, right)
```

**Comment by kmagiera** [comment]
@theoriginalgri this issue should be solved by #4481

On iOS at the moment you should st `textAlign` as a pat of `style` property

**Comment by kmagiera** [comment]
And the end goal is to unify that for android too, so after #4481 is merged we should only use textAlign as a part of style map as opposed to passing it as a component property directly
